### ISSUE TITLE
windowCount + share + take(1) does not complete correctly.

### ISSUE THREAD

**Issue by Dorus** [ReactiveX/rxjs#1698]
```Js
Rx.Observable.interval(500)
.windowCount(1)
.do(v => console.log('before share'))
.share()
.do(v => console.log('after share'))
.take(1)
.mapTo(0)
.subscribe(console.log.bind(console), null, () => console.log('completed'));
```

prints:

```text
"before share"
"after share"
0
"completed"
"before share"
"before share"
"before share"
"before share"
"before share"
...
```
(endless repetition).

I'm not sure what's going on here, but after reading `completed` i would expect the source to stop also. Replace `.take(1)` with `.take(0)` or `.take(2)` and it does stop as expected.

**Comment by trxcllnt** [comment]
resolved by #1705 and #1708

**Comment by kwonoj** [comment]
Closing by PR's checked in.
